components/molecule/accordion/src/index.js

```diff
@@ -4,23 +4,27 @@ import Tab from './Tab'
 
 const BASE_CLASS = 'sui-MoleculeAccordion'
 
-function MoleculeAccordion(props) {
-  const {children, defaultOpenedTabs = [], ...tabProps} = props
-
+const MoleculeAccordion = ({
+  children,
+  defaultOpenedTabs = [],
+  onToggleTab = () => {},
+  withAutoClose,
+  ...tabProps
+}) => {
   const initialOpenTabs = children.map((_, index) =>
     defaultOpenedTabs.includes(index)
   )
 
   const [openTabs, setOpenTabs] = useState(initialOpenTabs)
 
-  function onToggle(id) {
-    const {withAutoClose} = props
+  const _handleOnToggle = index => {
+    onToggleTab(index)
```

Comment:
The handler must return the Syntetic click event followed by an object with all the interesting elements to be returned

Reply:
It might be at the same lifecycle of other handlers or actions (L29-30)

Reply:
```js
// L20
const handleOnToggle = index => event => {
	let newOpenTabs = []
	if (withAutoClose) {
		newOpenTabs[index] = openTabs[index] ? undefined : true
	} else {
		newOpenTabs = [...openTabs]
		newOpenTabs[index] = newOpenTabs[index] ? undefined : true
	}
	onToggleTab(event, {index})
	setOpenTabs([...newOpenTabs])
}, [])
```

```js
// L38
onToggle={handleOnToggle(index)}
```